include/cadmium/modeling/dynamic_coupled.hpp

```diff
@@ -29,65 +29,96 @@
 
 #include <vector>
 #include <cassert>
-#include <cadmium/modeling/model.hpp>
+#include <cadmium/modeling/dynamic_model.hpp>
 #include <cadmium/modeling/dynamic_models_helpers.hpp>
+#include <boost/type_index.hpp>
 
 namespace cadmium {
-    namespace modeling {
+    namespace dynamic {
+        namespace modeling {
 
+            template<typename TIME>
+            class coupled : public cadmium::dynamic::modeling::model {
+            public:
+                Models _models;
+                Ports _input_ports;
+                Ports _output_ports;
+                EICs _eic;
+                EOCs _eoc;
+                ICs _ic;
 
-        template<typename TIME>
-        class dynamic_coupled : model {
+                std::string _id;
 
-            models_map _models;
-            ports_vector _input_ports;
-            ports_vector _output_ports;
-            EC_vector _eic;
-            EC_vector _eoc;
-            IC_vector _ic;
+                coupled() = delete;
 
-        public:
-            dynamic_coupled(
-                    models_map models,
-                    ports_vector input_ports,
-                    ports_vector output_ports,
-                    EC_vector eic,
-                    EC_vector eoc,
-                    IC_vector ic
-            ) {
-                assert(valid_ic_links(models, ic));
-                assert(valid_eic_links(models, input_ports, eic));
-                assert(valid_eoc_links(models, output_ports, eoc));
+                coupled(
+                        std::string id
+                ) :
+                        _id(id)
+                {
+                    assert(valid_ic_links(_models, _ic));
+                    assert(valid_eic_links(_models, _input_ports, _eic));
+                    assert(valid_eoc_links(_models, _output_ports, _eoc));
+                }
 
-                _models = models;
-                _input_ports = input_ports;
-                _output_ports = output_ports;
-                _eic = eic;
-                _eoc = eoc;
-                _ic = ic;
-            }
+                coupled(
+                        std::string id,
+                        Models models,
+                        Ports input_ports,
+                        Ports output_ports,
+                        EICs eic,
+                        EOCs eoc,
+                        ICs ic
+                ) :
+                        _id(id),
+                        _models(models),
+                        _input_ports(input_ports),
+                        _output_ports(output_ports),
+                        _eic(eic),
+                        _eoc(eoc),
+                        _ic(ic)
+                {
+                    assert(valid_ic_links(_models, _ic));
```

Comment:
I would say these can be thrown as domain errors

Reply:
Done! I changed all the asserts by domain errors here and in the dynamic_coordinator.hpp where i got the same scenario.